lib/workers/repository/extract/extract-fingerprint-config.ts

```diff
@@ -0,0 +1,67 @@
+import { mergeChildConfig } from '../../../config';
+import type { RenovateConfig } from '../../../config/types';
+import { get, getManagerList } from '../../../modules/manager';
+import type { WorkerExtractConfig } from '../../types';
+
+export interface FingerprintExtractConfig {
+  managerList: Set<string>;
+  managers: WorkerExtractConfig[];
+}
+
+function getFilteredManagerConfig(
+  config: RenovateConfig,
+  manager: string
+): WorkerExtractConfig {
+  const language = get(manager, 'language');
+  let mergedConfig = {} as RenovateConfig;
+  if (language) {
+    mergedConfig = mergeChildConfig(config, config[language] as any);
+  }
+
+  mergedConfig = mergeChildConfig(mergedConfig, config[manager] as any);
+  return {
+    manager,
+    npmrc: mergedConfig.npmrc,
+    npmrcMerge: mergedConfig.npmrcMerge,
+    enabled: mergedConfig.enabled,
+    ignorePaths: mergedConfig.ignorePaths ?? [],
+    includePaths: mergedConfig.includePaths ?? [],
+    skipInstalls: mergedConfig.skipInstalls,
+    registryAliases: mergedConfig.registryAliases,
+    fileMatch: mergedConfig.fileMatch ?? [],
+    fileList: [],
+  };
+}
+
+export function generateFingerprintConfig(
+  config: RenovateConfig
+): FingerprintExtractConfig {
+  const finalConfig = {} as FingerprintExtractConfig;
+
+  const managerExtractConfigs: WorkerExtractConfig[] = [];
+  let managerList = new Set(getManagerList());
+  const { enabledManagers } = config;
+  if (enabledManagers?.length) {
+    managerList = new Set(enabledManagers);
+  }
+  finalConfig.managerList = managerList;
+
+  for (const manager of managerList) {
+    if (manager === 'regex') {
+      for (const managerConfig of config.regexManagers ?? []) {
+        managerExtractConfigs.push({
+          manager,
+          fileList: [],
+          enabled: true,
+          ...managerConfig,
+        });
+      }
+    } else {
+      managerExtractConfigs.push(getFilteredManagerConfig(config, manager));
```

Comment:
I think the problem is that you are not merging the original `config`. For example it might commonly include `ignorePaths` at the top level instead of inside a manager or language. Instead you should do this:

- merge the language config (if found) on top of config
- merge the manager config (if found) on top of the above
- perform filtering